e2e/migrations/migrations_test.go

```diff
@@ -0,0 +1,176 @@
+package e2e
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func isSuccess(execCmd *exec.Cmd, retries int, keyword string) (bool, string) {
+	numAttempts := retries
+	success := false
+	var out bytes.Buffer
+
+	for numAttempts > 0 {
+		time.Sleep(2 * time.Second)
+
+		cmd := *execCmd
+		cmd.Stdout = &out
+
+		err := cmd.Run()
+		if err != nil {
+			return false, ""
+		}
+
+		success = (out.String() != "" && !strings.Contains(out.String(), keyword))
+		if success {
+			return success, out.String()
+		}
+
+		out.Reset()
+		numAttempts -= 1
+	}
+
+	return success, out.String()
+}
+
+func allNomadNodesAreReady(retries int) (bool, string) {
+	cmd := exec.Command("nomad", "node-status")
+	return isSuccess(cmd, retries, "initializing")
+}
+
+func jobIsReady(retries int, jobName string) (bool, string) {
+	cmd := exec.Command("nomad", "job", "status", jobName)
+	return isSuccess(cmd, retries, "pending")
+}
+
+// requires nomad executable on the path
+func startCluster(clusterConfig []string) (func(), error) {
+	cmds := make([]*exec.Cmd, 0)
+
+	for _, agentConfig := range clusterConfig {
+		cmd := exec.Command("nomad", "agent", "-config", agentConfig)
+		err := cmd.Start()
+
+		if err != nil {
+			return func() {}, err
+		}
+
+		cmds = append(cmds, cmd)
+	}
+
+	f := func() {
+		for _, cmd := range cmds {
+			cmd.Process.Kill()
```

Comment:
You have to wait on the process or they will be zombied.

Reply:
I'm not sure the test would finish if we used process.Wait, as this is creating a running client/server. See https://golang.org/pkg/os/#Process.Wait

If each test runs a cluster in a separate docker container, we should be able to finish without proper cleanup.

Reply:
You can call wait after you call kill, but I see your point